File: frontend/src/types/agent.ts
export type AgentName =
  | 'extraction'
  | 'industry'
  | 'collateral'
  | 'guarantor'
  | 'packaging'
  | 'financial'
  | 'risk'
  | 'coordination'
  | 'review'
  | 'deckr'

export interface AgentInfo {
  name: AgentName
  display_name: string
  mode: 'generate' | 'chat'
  conversational: boolean
}

export interface AgentMessage {
  role: 'user' | 'agent'
  content: string
  agent_name?: AgentName    // present on agent replies
  saved_to?: string         // populated if agent saved output to workspace
  timestamp: number
}

export interface AgentRequest {
  message: string
  agent_name: AgentName
  session_id: string
  messages: { role: string; content: string }[]
  save_to_workspace?: boolean
  save_path?: string
  action_type?: string      // routes to a specific action prompt template (e.g. "analyze_leverage")
  tools?: object[]          // reserved for Phase 15 tool calling
}

export interface AgentResponse {
  reply: string
  agent_name: AgentName
  session_id: string
  saved_to: string | null
  saved_files?: string[] | null   // all files written during a multi-save run
}

// ---------------------------------------------------------------------------
// Pipeline types — Phase 13
// ---------------------------------------------------------------------------

export interface PipelineRequest {
  session_id: string
  message?: string
}

export type PipelineEventType =
  | 'pipeline_start'
  | 'step_start'
  | 'step_done'
  | 'step_error'
  | 'pipeline_complete'

export interface PipelineEvent {
  type: PipelineEventType
  // pipeline_start
  total?: number
  // step_start / step_done / step_error
  agent?: string
  display_name?: string
  step?: number
  // step_done
  saved_to?: string | null
  elapsed_ms?: number
  reply_preview?: string
  // step_error
  error?: string
  // pipeline_complete
  steps_done?: number
  steps_failed?: number
  total_elapsed_ms?: number
}

export interface PipelineStepState {
  agent: string
  display_name: string
  status: 'pending' | 'running' | 'done' | 'error'
  saved_to?: string | null
  elapsed_ms?: number
  error?: string
}

File: backend/models/agent.py
from pydantic import BaseModel


class AgentRequest(BaseModel):
    message: str
    agent_name: str
    session_id: str                   # UUID — required from Day 1; becomes Orchestrate memory scope in Phase 12
    messages: list[dict]              # full prior conversation history sent with every request
    save_to_workspace: bool = False
    save_path: str | None = None
    action_type: str | None = None    # routes to a specific action prompt template (e.g. "analyze_leverage")
    tools: list[dict] | None = None   # reserved for Phase 13 tool calling


class AgentResponse(BaseModel):
    reply: str
    agent_name: str
    session_id: str
    saved_to: str | None = None
    saved_files: list[str] | None = None   # all files written during a multi-save run
